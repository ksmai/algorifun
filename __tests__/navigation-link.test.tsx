import React from 'react';
import renderer from 'react-test-renderer';

import NavigationLink from '../src/components/navigation-link';

describe('NavigationLink', () => {
    it('renders GastbyLink correctly', () => {
        const tree = renderer
            .create(<NavigationLink to="/">Home</NavigationLink>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders <a> correctly', () => {
        const tree = renderer
            .create(<NavigationLink href="https://www.google.com/" target="_blank">Google</NavigationLink>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
