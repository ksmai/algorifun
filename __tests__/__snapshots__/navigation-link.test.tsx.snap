// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationLink renders <a> correctly 1`] = `
<a
  css={
    Object {
      "map": undefined,
      "name": "k49009",
      "next": undefined,
      "styles": "
    text-decoration: none;
    &:hover {
        color: #F48FB1;
    }
",
      "toString": [Function],
    }
  }
  href="https://www.google.com/"
  target="_blank"
>
  Google
</a>
`;

exports[`NavigationLink renders GastbyLink correctly 1`] = `
<a
  css={
    Object {
      "map": undefined,
      "name": "k49009",
      "next": undefined,
      "styles": "
    text-decoration: none;
    &:hover {
        color: #F48FB1;
    }
",
      "toString": [Function],
    }
  }
  href="/"
>
  Home
</a>
`;
